File: benchmarks/bench_fib.js
// Benchmark: Recursive Fibonacci fib(30) — Node.js
// Run: node bench_fib.js

function fib(n) {
  if (n <= 1) return n;
  return fib(n - 1) + fib(n - 2);
}

const start = performance.now();
const result = fib(30);
const elapsed = Math.round(performance.now() - start);
console.log(`fib(30) = ${result}`);
console.log(`Time: ${elapsed}ms`);

File: benchmarks/bench_fib.go
// Benchmark: Recursive Fibonacci fib(30) — Go
// Run: go run bench_fib.go
package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n <= 1 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	start := time.Now()
	result := fib(30)
	elapsed := time.Since(start).Milliseconds()
	fmt.Printf("fib(30) = %d\n", result)
	fmt.Printf("Time: %dms\n", elapsed)
}
